core/metrics-core-service/src/test/java/org/hawkular/metrics/core/service/MetricsServiceITest.java

```diff
@@ -288,55 +288,50 @@ public void createAndFindMetricsWithTags() throws Exception {
-
-        Metric<Double> mA = new Metric<>(new MetricId<>(tenantId, GAUGE, "mA"),
-                ImmutableMap.of("hostname", "webfin01"), 24);
-        Metric<Double> mB = new Metric<>(new MetricId<>(tenantId, GAUGE, "mB"),
-                ImmutableMap.of("hostname", "webswe02"), 24);
-        Metric<Double> mC = new Metric<>(new MetricId<>(tenantId, GAUGE, "mC"), ImmutableMap.of("hostname",
-                                                                                              "backendfin01"),
-                                         24);
-        Metric<Double> mD = new Metric<>(new MetricId<>(tenantId, GAUGE, "mD"), ImmutableMap.of("hostname",
-                                                                                              "backendswe02"),
-                                         24);
-        Metric<Double> mE = new Metric<>(new MetricId<>(tenantId, GAUGE, "mE"), ImmutableMap.of("owner", "hede"),
-                                         24);
-        Metric<Double> mF = new Metric<>(new MetricId<>(tenantId, GAUGE, "mF"), ImmutableMap.of("owner", "hades"),
-                                         24);
-        Metric<Double> mG = new Metric<>(new MetricId<>(tenantId, GAUGE, "mG"), ImmutableMap.of("owner", "had"),
-                                         24);
-
-        // Create the availabilities
-        Metric<AvailabilityType> a1 = new Metric<>(new MetricId<>(tenantId, AVAILABILITY, "a1"),
-                                                   ImmutableMap.of("a1","4"), 24);
+        List<Metric> metricsToAdd = new ArrayList<>(ids.size());
+        for (int i = 0; i < ids.size(); i++) {
+            metricsToAdd.add(new Metric(ids.get(i), maps.get(i), 24));
+        }
 
         // Insert gauges
-        metricsService.createMetric(m1).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(m2).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(m3).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(m4).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(m5).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(m6).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(a1).toBlocking().lastOrDefault(null);
-
-        metricsService.createMetric(mA).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(mB).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(mC).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(mD).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(mE).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(mF).toBlocking().lastOrDefault(null);
-        metricsService.createMetric(mG).toBlocking().lastOrDefault(null);
+        for (Metric m : metricsToAdd) {
+            metricsService.createMetric(m).toBlocking().lastOrDefault(null);
```

Comment:
This should not use a for-clause, instead an observable. Like: Observable.from(metricsToAdd).subscribe(m -> metricsService.createMetric(m)).. etc add blocking mechanism at the end.